src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -363,12 +361,13 @@ using MergeRegexPatternAndFlagsExpression =
     StringExpressionImpl<2, decltype(mergeFlagsIntoRegex), LiteralFromIdGetter>;
 
 [[maybe_unused]] auto replaceImpl =
-    [](std::optional<std::string> input, const std::shared_ptr<RE2>& pattern,
+    [](std::optional<ad_utility::triple_component::Literal> s,
+       const std::shared_ptr<RE2>& pattern,
        const std::optional<std::string>& replacement) -> IdOrLiteralOrIri {
-  if (!input.has_value() || !pattern || !replacement.has_value()) {
+  if (!s.has_value() || !pattern || !replacement.has_value()) {
     return Id::makeUndefined();
   }
-  auto& in = input.value();
+  std::string in(asStringViewUnsafe(s.value().getContent()));
```

Comment:
Currently the output doesn't get the correct datatype or language tag, hence the conformance test still fails.
the `toLiteral` function in line 378 either has to be replaced or extended.